src/cli/global/sync.rs

```diff
@@ -28,16 +28,20 @@ pub async fn execute(args: Args) -> miette::Result<()> {
     }
 
     let mut errors = Vec::new();
+    let mut has_changed = false;
```

Comment:
The state changes variable changes should still be reported once (to report on pruned envs).

Reply:
You are right, nice find!

I fixed the code and added a test for this.